eslint.config.ts

```diff
@@ -51,7 +53,7 @@ const config: ReturnType<typeof tseslint.config> = tseslint.config(
   //#endregion
 
   //#region typescript-eslint
-  ...tseslint.configs.strictTypeChecked,
+  ...(tseslint.configs.strictTypeChecked as ConfigWithExtendsArray),
```

Comment:
[According to the ts-eslint docs](https://typescript-eslint.io/users/configs/#strict-type-checked), this config can be provided without spreading. Then the type assertion should not be required:


```suggestion
  tseslint.configs.strictTypeChecked,
```

The resulting configs are the same from my quick testing using eslint's `--print-config` flag.

Reply:
That way we don't need the extra dependency.

Reply:
The extra dependency is needed anyway, because this is from what eslint's `defineConfig` is coming from

Reply:
What? Why that? `defineConfig` is exposed from the `eslint` package under the `eslint/config` path [as by eslint's offical documentation](https://eslint.org/docs/latest/use/configure/configuration-files). 🤔

Reply:
This might be new 😲

Reply:
The PR https://github.com/eslint/eslint/pull/19487 introduced the `definedConfig` function in the "eslint" package, which was first introduced [v9.22.0 of ESLint](https://github.com/eslint/eslint/releases/tag/v9.22.0) - so about 7 months ago as of writing this comment.

---

I was curious since the way I described it has "always" been the way I've setup ESLint in my enterprise work environment (since the migration to v9 was approved). So this is not a "your way is wrong, my  way is right" kind of thing, just an interesting to know information. :)